benchmark/projection/args.cpp

```diff
@@ -0,0 +1,82 @@
+#include "include/args.hpp"
+#include <climits>
+#include <cstdio>
+#include <cstdlib>
+#include <unistd.h>
+static int get_int(char *str) {
+  char *endptr = nullptr;
+  auto n = strtol(str, &endptr, 10);
+  if (n == LONG_MIN || n == LONG_MAX) {
+    perror("Argument under/overflow");
+  }
+  if (endptr == str) {
+    // NOTE: I think this is not safe using printf with user input
+    // so don't use code for anything "real"
+    fprintf(stderr, "%s is not valid value\n", str);
+    exit(1);
+  }
+  if (n <= 0) {
+    fprintf(stderr, "argument values must be positive\n");
+    exit(1);
+  }
+  // TODO: should I just change to long everywhere?
+  if (n > INT_MAX) {
+    fprintf(stderr, "Max argument value is %d\n", INT_MAX);
+    exit(1);
+  }
+  return (int)n;
+}
+
+void print_options(CmdArgs const &arg) {
+  printf("Options:\n"
+         "number of cells %d\n"
+         "number of active cells (have particles) %d\n"
+         "max particles per cell %d\n"
+         "min particles per cell %d\n",
+         arg.ncell, arg.active_cell, arg.max_per_cell, arg.min_per_cell);
+}
+
+CmdArgs get_args(int argc, char **argv, bool print) {
+  CmdArgs arg;
+  int opt = -1;
+  while ((opt = getopt(argc, argv, "m:n:a:f:t:l:hse")) != -1) {
```

Comment:
How portable is getopt going to be? We already have have boost as a dependency (as does nektar) so we could use boost::program_options instead?

Reply:
Should work on mac it's POSIX don't have a mac to test it though. No idea with windows but probably not - are we supporting windows? I will take a look at boost.  Looking at it I think the string ("m:n:a....")  has too many options (lazy copy and paste) so needs fixing anyway.